### ISSUE TITLE
`CString`s don't escape and fail to link as triple quoted strings

### ISSUE THREAD

**Issue by lolgab** [scala-native/scala-native#1737]
This is an example of a `CString` with escapes
```scala
val s: CString = c"{\"value\": {\"greeting\": \"Hello world!\"}}"
```
If you try with a triple quoted string you get a linking error:
```scala
val s: CString = c"""{"value": {"greeting": "Hello world!"}}"""
```
```
[error] target/scala-2.11/native/6.ll:221:63: error: constant expression type mismatch
[error] @"_SM7__constG3213" = private unnamed_addr constant [40 x i8] c"{"value": {"greeting": "Hello world!"}}\00"
[error]                                                               ^
[error] 1 error generated.
```

**Comment by sjrd** [comment]
Thanks for the report.

**Comment by lolgab** [comment]
Closing this since it was solved in #1855
